### ISSUE TITLE
Multiple ZSH_CUSTOM directories

### ISSUE THREAD

**Issue by bdwalton** [ohmyzsh/ohmyzsh#11095]
### If the feature solves a problem you have, specify it here.

I have a default environment that I use on all of my machines. But I also have a private set of configuration that I use at work and can't store in my public dotfiles. I've hacked up a plugin that I carry in my public repo which can pull in a an additional set of plugins but it's not very clean.

### Describe the proposed feature.

 I'd love to think about making a ZSH_CUSTOM_DIRS _array_ that could be leveraged to more cleanly allow multiple custom directories, with precedence based on array ordering. If you're open to such a feature, I'd work with you to contribute the code.

(I'd also explore the idea of doing this as OMZ_CUSTOM to try out one of the other proposed feature requests - renaming the variable prefix away from ZSH.)

**Comment by daltinn** [comment]
@bdwalton do you try to use Antigen for solve this?
I have same problem. After reading about antigen, I think it may be more complex solving the problem because Antigen also updating plugins.

**Comment by bdwalton** [comment]
No, I just wrote a plugin that resides the main customdir that does what I
need. It's not beautiful. But it's not hideous overall. Definitely meets my
needs.

I did look at adding the support for this feature and would be happy to
follow-up on that if there is more than just me interested.

**Comment by rwmitchell** [comment]
Just as a suggestion, I don't put anything under 'custom' except symlinks to plugins kept elsewhere.

Given the frequent suggestion on discord recommending people to 'rm -rf $ZSH' to fix any problems they are having with OMZ (which is completely barbaric) I avoid putting anything important there.

**Comment by bdwalton** [comment]
Well, I guess people make poor choices , but we shouldn't engineer
solutions to lowest common denominator... In all cases, anyway. 😀

**Comment by bdwalton** [comment]
I think custom is a natural concept in omz though. Allowing those to be
sourced from multiple locations is also pretty natural, no?

**Comment by rwmitchell** [comment]
By putting symlinks to the real location, you can still do that.  I really don't understand your issue.
If OMZ does get removed, all of your custom files still exist locally, somewhere else on your system.

**Comment by elanora96** [comment]
I was disappointed to find that multiple ZSH_CUSTOM directories aren't a thing. I got around the restriction by making a `other_plugin_folders` array. Then, the directories are symlinked into `$ZSH/custom/plugins` if their basenames do not already exist there. Place this before sourcing `$ZSH/oh-my-zsh.sh`

#### .zshrc
```sh
# Until oh-my-zsh allows multiple ZSH_CUSTOM folders, here's a workaround

other_plugins_folders=("/usr/share/zsh/plugins" "/home/elanora96/Projects/plugins" "/Foo/Bar/plugins")

for dir in "${other_plugins_folders[@]}"; do
  if [[ -d "$dir" ]]; then
    for sdir in "$dir"/*; do
      if [[ ! -e "$ZSH/custom/plugins/$(basename "$sdir")" ]]; then
        ln -s "$sdir" "$ZSH/custom/plugins"
      fi
    done
  fi
done
```

* Do note that this snippet is not a true multiple `$ZSH_CUSTOM` replacement, its scope is just the `$ZSH/custom/plugins` folder as that is all I needed.

**Comment by mattmc3** [comment]
I published a project that addresses this issue: https://github.com/mattmc3/omz-plus

TLDR - in your .zshrc add the following:

```zsh
# Use either ZSH_CUSTOM for a single custom directory...
# ZSH_CUSTOM=/path/to/new-custom-folder
# ...OR... use OMZ PLUS! $zsh_custom variable for multiple
zsh_custom=(
    ${ZDOTDIR:-$HOME}/.zsh_custom
    ${ZDOTDIR:-$HOME}/.zsh_custom_work
)

##### START OMZ PLUS! ##################################################################
export OMZ_PLUS=${ZDOTDIR:-$HOME}/.omz-plus
[ -d "$OMZ_PLUS" ] || git clone https://github.com/mattmc3/omz-plus $OMZ_PLUS
source $OMZ_PLUS/omz-plus.sh
##### END OMZ PLUS! ####################################################################

# Source oh-my-zsh.sh immediately after omz-plus.sh
source $ZSH/oh-my-zsh.sh
```

**Comment by robbyrussell** [comment]
Sorry, everyone — three years and nobody from the project weighed in. @bdwalton, you said back in 2023 you'd build it if others were interested, and three people said they were. We should have answered then.

Our position: Oh My Zsh manages one custom directory, and not the rest of your dotfiles. Supporting a list means an N-way precedence chain for themes, changes at around twenty places across `oh-my-zsh.sh` and `lib/cli.zsh`, and a new rule for which directory `omz plugin enable` writes into — for something symlinks already do.

Symlinks genuinely work, and I checked all three paths:

```zsh
ln -s ~/work-dotfiles/plugins/foo        "$ZSH_CUSTOM/plugins/foo"
ln -s ~/work-dotfiles/mytheme.zsh-theme  "$ZSH_CUSTOM/themes/mytheme.zsh-theme"
ln -s ~/work-dotfiles/extra.zsh          "$ZSH_CUSTOM/extra.zsh"
```

Plugin directories, themes and loose `.zsh` files all resolve through symlinks, because the loading code uses `test -f` and plain globs. And to @rwmitchell's point about `rm -rf $ZSH` eating your work: point `ZSH_CUSTOM` at a directory *outside* `$ZSH` that holds those symlinks, and that risk goes away too.

If you want the real array version, @mattmc3's [omz-plus](https://github.com/mattmc3/omz-plus) does it as a shim you source before Oh My Zsh — that's the right layer for it.

Closing so it isn't sitting in limbo any longer.

🤖 Co-authored with Claude Code
